### DISCUSSION TITLE
[Feature]: Add support for visionOS full screen resize on immersive spaces

### DISCUSSION THREAD

**Discussion by j6alvarez** [TheWidlarzGroup/react-native-video#4987] [Ideas]
### What problem are you trying to solve?

I am developing a react-native-visionos app and using the immersive space I cannot resize the screen while on immersive mode. This seems possible on native environment on other apps.

The resize and movement of the screen would be a nice feature for the videoplayer to have.

Here is a screenshot of the app on immersive mode and fullscreen:
![image](https://github.com/user-attachments/assets/3d4c8182-355c-4057-a1b3-5f4c4b06b97c)

And here is a screenshot of the app on fullscreen without an immersive space:
![image](https://github.com/user-attachments/assets/f7aacd5f-8c0e-4724-a8a2-eb5a6ed962d9)

### Proposed solution

Add support for visionOS full screen resize on immersive spaces, allowing the video player screen to be resized and moved while on immersive mode.

I have look into some documentation and I am not certain but I think what would be needed is to implement the use of AVPlayerViewController in the RCTVideo.swift file for visionOS. According to documentation this AVPlayerViewController is the one that gets on the [DockingRegionComponent](https://developer.apple.com/documentation/RealityKit/DockingRegionComponent) being unable to move it or resize it. Seems that per default the videoplayer is being placed on this section and we have no way to edit this.

### Platforms

visionOS

### Additional context

Here are other resources:
- https://developer.apple.com/documentation/visionos/destination-video
- https://developer.apple.com/design/human-interface-guidelines/immersive-experiences/
- https://developer.apple.com/documentation/avkit/adopting-the-system-player-interface-in-visionos

[a comment by an automated account is not shown]

**Comment by KrzysztofMoch** [comment]
Hey @j6alvarez, It seems we cannot resize fullscreen videos with `react-native-video`. It's probably possible with SwiftUI using `RealityKit` or `SceneKit`, but we probably can't do it within library.

**Comment by j6alvarez** [comment]
Hello @KrzysztofMoch , thanks for the answer! I see what you mean. But do you have an idea of how it is possible to do when you are not on an immersive space to resize? In the example, I can resize the fullscreen video (second screenshot). What I am asking is to be able to have the behavior of the player without an immersive space (second screenshot) inside an immersive space (first screenshot) to be able to resize and move it in the immersive space.

**Comment by moskalakamil** [comment]
Hey @j6alvarez, do you still have this chalange? Did something changed?

**Comment by j6alvarez** [comment]
Hello @moskalakamil, the situation has not changed. We have this issue in backlog since it's not a priority for now. But it stills being an issue. I'll be happy to provide more context if needed. I think this would be a nice feature to have for react-native-video in visionOS

**Comment by moskalakamil** [comment]
I get it @j6alvarez, I completely agree. Personally, I haven’t delved into the topic, but as @KrzysztofMoch mentioned, there is a possibility that it can’t be done in the library and it's not currently on the roadmap. However, if this is very valuable for you, please reach out to us ([hi@thewidlarzgroup.com](mailto:hi@thewidlarzgroup.com)) – we could investigate the matter further and offer commercial support

**Comment by TWG-AI** [comment]
Thanks for the detailed request! Window/video resize in visionOS immersive spaces is a platform capability worth supporting — on native AVPlayer it works but react-native-video doesn't expose it yet. We've noted this for the team to evaluate.
